x/attribute/keeper/keeper_test.go

```diff
@@ -740,3 +740,54 @@ func (s *KeeperTestSuite) TestIterateRecord() {
 	})
 
 }
+
+func (s *KeeperTestSuite) TestPopulateAddressAttributeNameTable() {
+	store := s.ctx.KVStore(s.app.GetKey(types.StoreKey))
+
+	example1Attr := "example.one"
+	exampleAttr1 := types.NewAttribute(example1Attr, s.user1, types.AttributeType_String, []byte("test1"))
+	exampleAttr2 := types.NewAttribute(example1Attr, s.user1, types.AttributeType_String, []byte("test2"))
+	exampleAttr3 := types.NewAttribute(example1Attr, s.user1, types.AttributeType_String, []byte("test3"))
+	exampleAttr4 := types.NewAttribute(example1Attr, s.user2, types.AttributeType_String, []byte("test4"))
+	s.Assert().NoError(s.app.NameKeeper.SetNameRecord(s.ctx, example1Attr, s.user1Addr, false), "name record should save successfully")
+	s.Assert().NoError(s.app.AttributeKeeper.SetAttribute(s.ctx, exampleAttr1, s.user1Addr), "should save successfully")
+	s.Assert().NoError(s.app.AttributeKeeper.SetAttribute(s.ctx, exampleAttr2, s.user1Addr), "should save successfully")
+	s.Assert().NoError(s.app.AttributeKeeper.SetAttribute(s.ctx, exampleAttr3, s.user1Addr), "should save successfully")
+	s.Assert().NoError(s.app.AttributeKeeper.SetAttribute(s.ctx, exampleAttr4, s.user1Addr), "should save successfully")
+
+	example2Attr := "example.two"
+	example2Attr1 := types.NewAttribute(example2Attr, s.user1, types.AttributeType_String, []byte("test1"))
+	example2Attr2 := types.NewAttribute(example2Attr, s.user2, types.AttributeType_String, []byte("test2"))
+	s.Assert().NoError(s.app.NameKeeper.SetNameRecord(s.ctx, example2Attr, s.user1Addr, false), "name record should save successfully")
+	s.Assert().NoError(s.app.AttributeKeeper.SetAttribute(s.ctx, example2Attr1, s.user1Addr), "should save successfully")
+	s.Assert().NoError(s.app.AttributeKeeper.SetAttribute(s.ctx, example2Attr2, s.user1Addr), "should save successfully")
+
+	// Clear the kv store of all address look up prefixes
+	// This is because the SetAttribute call would have populated it in the test setup
+	it := sdk.KVStorePrefixIterator(store, types.AttributeKeyPrefixAddrLookup)
+	for ; it.Valid(); it.Next() {
+		store.Delete(it.Key())
+	}
+
```

Comment:
we should have an assert here that the key is not found before we recreate them all

Reply:
That's a good check.  Done.